subprojects/core/src/integTest/groovy/org/gradle/api/TaskContainerIntegrationTest.groovy

```diff
@@ -16,23 +16,37 @@
 
 package org.gradle.api
 
-
+import groovy.transform.SelfType
 import spock.lang.Issue
 
-class TaskContainerIntegrationTest extends AbstractDomainObjectContainerIntegrationTest {
-    @Override
+@SelfType(AbstractDomainObjectContainerIntegrationTest)
+trait AbstractTaskContainerIntegrationTest {
     String makeContainer() {
         return "tasks"
     }
 
-    @Override
     String getContainerStringRepresentation() {
         return "task set"
     }
 
     static String getContainerType() {
         return "DefaultTaskContainer"
     }
+}
+
+class QueryAndMutationTaskContainerIntegrationTest extends AbstractQueryAndMutationDomainObjectContainerIntegrationTest implements AbstractTaskContainerIntegrationTest {
```

Comment:
I wonder how to organize these classes into files. I think it would be good to have a separate file per class which actually does something. For these empty subclasses, I am not sure. Though I wouldn't call the file then like any of those classes. WDYT?

Reply:
Let's leave it like is then.